quantum/gate/ir/CommonGates.hpp

```diff
@@ -301,6 +301,49 @@ class U : public Gate {
   DEFINE_VISITABLE()
 };
 
+// The R_phi(theta) rotation from the Methods section of
+// https://www.nature.com/articles/nature18648
+class Rphi: public Gate {
+public:
+  Rphi()
+      : Gate("Rphi",
+             std::vector<InstructionParameter>{InstructionParameter(0.0),
+                                               InstructionParameter(0.0)}) {}
+  Rphi(std::size_t qbit, std::vector<xacc::InstructionParameter> params)
+      : Gate("Rphi", std::vector<std::size_t>{qbit}, params) {}
+  Rphi(std::size_t qbit, double phi, double theta)
+      : Gate("Rphi", std::vector<std::size_t>{qbit},
+             std::vector<InstructionParameter>{InstructionParameter(phi),
+                                               InstructionParameter(theta)}) {}
+
+  const int nRequiredBits() const override { return 1; }
+
+  DEFINE_CLONE(Rphi)
+  DEFINE_VISITABLE()
+};
+
+// The XX(alpha) entangling gate from the Methods section of
+// https://www.nature.com/articles/nature18648
+// (We call the parameter alpha instead of chi_ij)
+class XX : public Gate {
```

Comment:
This name is inconsistent with RZZ (below), unfortunately. RZZ is basically the same thing, the only difference is for XX the definition is exp(-i(X⊗X)*θ/2) instead of exp(-i(Z⊗Z)*θ/2) for RZZ.

I'm guessing RZZ is called that [because qiskit calls it RZZ](https://qiskit.org/documentation/stubs/qiskit.circuit.library.RZZGate.html) (I could be wrong though). However, I haven't seen the XX gate called RXX anywhere and failed to find examples on Google or Google Scholar... [except qiskit](https://qiskit.org/documentation/stubs/qiskit.circuit.library.RXXGate.html). Not sure what the right call on should be on naming this in XACC, but just disclosing this